aspnetcore/grpc/performance.md

```diff
@@ -0,0 +1,112 @@
+---
+title: Performance best practices
+author: jamesnk
+description: Learn the best practices for building high-performance gRPC services.
+monikerRange: '>= aspnetcore-3.0'
+ms.author: jamesnk
+ms.date: 08/23/2020
+no-loc: ["ASP.NET Core Identity", cookie, Cookie, Blazor, "Blazor Server", "Blazor WebAssembly", "Identity", "Let's Encrypt", Razor, SignalR]
+uid: grpc/performance
+---
+# Performance best practices
+
+By [James Newton-King](https://twitter.com/jamesnk)
+
+gRPC is designed for high-performance services. This document explains how to get the best performance possible from gRPC.
+
+## Reuse channel
+
+A gRPC channel should be reused when making gRPC calls. Reusing a channel allows calls to be multiplexed through an existing HTTP/2 connection.
+
+If a new channel is created for each gRPC call then the amount of time it takes to complete can increase significantly. Each call will require multiple network round-trips between the client and the server to create an HTTP/2 connection:
+
+1. Opening a socket
+2. Establishing TCP connection
+3. Negotiating TLS
+4. Starting HTTP/2 connection
+5. Making the gRPC call
+
+Channels are safe to share and reuse between gRPC calls:
+
+* gRPC clients are created with channels. gRPC clients are lightweight objects and don't need to be cached or reused.
+* Multiple gRPC clients can be created from a channel, including different types of clients.
```

Comment:
Should we mention the max streams per connection limit? We've had several issues filed when that limit is reached so we might want to at least mention it here.